Adding Dynatrace support

oybed opened the pull request:
### What does this PR do?
This PR introduces support for Dynatrace templates to be used to deploy Dynatrace monitoring of an OpenShift cluster. 

The original work can be found at the following URL, but it doesn't support deployment as templates: https://github.com/Dynatrace/dynatrace-oneagent-operator/tree/master/deploy

### How should this be tested?
Multiple ways the templates can be tested. One approach is to utilize the `openshift-applier` inventory part of this PR. 

### Is there a relevant Issue open for this?
N/A

### Other Relevant info, PRs, etc.
https://github.com/redhat-cop/monitoring/pull/1

### People to notify
cc: @redhat-cop/monitoring

makentenza commented:
@oybed Some initial comments after testing this:

- I'm missing a README file with instructions to deploy
- Once deployed the pod has not enough privileges to get objects from the cluster. I've just checked the resultant deployment and pod and these are using the 'default' service account from the project which has no extended privileges on the cluster.

oybed commented:
@makentenza yes, sorry, these are knows things to work on as mentioned in chat the other day. I updated the title of the PR to appropriately reflect that this isn't complete yet.

makentenza commented:
Ok, make sense @oybed. Will check again once is completed. Another quick note: Why are we using 2 different Tokens here? I think there is not really a requirement to do so

oybed commented:
Updated the PR with the latest definitions + fixed the permission issues. 

@makentenza please take another look. 

@sabre1041 FYI

BTW: added a README with instructions - it can be read here: https://github.com/oybed/monitoring/blob/dynatrace/dynatrace/openshift/README.md

oybed commented:
@makentenza re: the 2 tokens - that's probably more of a question for Dynatrace, but you can see their use of the 2 tokens (the same way we're using them) in their docs here: https://github.com/Dynatrace/dynatrace-oneagent-operator#dynatrace-oneagent-operator

makentenza commented:
thanks @oybed Will test this with the Labs Customer the following 2 weeks so I'll add myself as a reviewer for this

makentenza commented on dynatrace/openshift/README.md:
```diff
@@ -0,0 +1,46 @@
+5. Modify parameters in `params/dynatrace-secret.params` to set the following config options:
+ * Dynatrace API Token (in base64 encoding - see below)
+ * Dynatrace PaaS Token (in base64 encoding - see below)
+
```
Why are we making `base64` a requirement? Can we just use `stringData` instead on the Secret for the API Token? This is not encrypted anyway, so we could save some steps and explanations.

makentenza replied:
We should add a link on how to create those Tokens from the [Dynatrace Documentation](https://www.dynatrace.com/support/help/shortlink/openshift-applications#generate-paas-token)

makentenza commented on dynatrace/openshift/README.md:
````diff
@@ -0,0 +1,46 @@
+
+Run the [openshift-applier](https://github.com/redhat-cop/openshift-applier) to deprovision the Dynatrace Stack and delete all the objects created during the deploying phase
+```
+> ansible-playbook -i ./inventory galaxy/openshift-applier/playbooks/openshift-cluster-seed.yml -e 'provision=false'
````
Even this is removing all the OpenShift objects, the installation is leaving behind a lot of files on every Node. Maybe something to report back to Dynatrace?

```
[root@ip-10-20-1-55 oneagent]# pwd
/opt/dynatrace/oneagent
[root@ip-10-20-1-55 oneagent]# du -hs .
894M	.
[root@ip-10-20-1-55 oneagent]# find ./ | wc -l
4460
```

oybed replied:
Sure, please submit the request to Dynatrace (see link in original comment for this PR)

oybed commented:
@makentenza the 2 tokens is per the instructions provided by dynatrace (see link in my earlier comment) and I know there are 2 places these are generated in their solution, so I'm going to leave this as-is per the instructions from them.

oybed commented:
BTW: re: the 2 tokens ... the documentation you linked is referencing "application only" docs. We are monitoring both the platform and the applications with the implementation in this PR, so I suspect that's why the 2 tokens are needed.

makentenza commented:
@oybed yes, I can see the difference, you are right. Just for clarification, should we open an issue on their `dynatrace-oneagent-operator` GitHub repository? I can do it myself if you want

oybed commented:
@makentenza Please do: https://github.com/Dynatrace/dynatrace-oneagent-operator/blob/master/README.md#openshift-2

makentenza commented:
@oybed Just created 3 issues to address all these questions/problems

oybed commented:
@makentenza please see updated PR for changes incorporated based on your feedback. Also, please see my comment on one of the 3 issues, I think that helps explain the 2 tokens.

https://github.com/Dynatrace/dynatrace-oneagent-operator/issues/43#issuecomment-415878745

makentenza commented:
Great, thanks @oybed, just checked your comments on that.

This LGTM now!

sabre1041 commented:
LGTM
